### ISSUE TITLE
Rule idea: selector-nest-* (selector type)

### ISSUE THREAD

**Issue by kristerkari** [kristerkari/stylelint-scss#299]
Pull request #298 implemented the `selector-nest-combinators` rule to check that selector combinators are nested.

Based on this list https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Selectors, there could also be these additional rules:

- `selector-nest-types`
- `selector-nest-classes`
- `selector-nest-ids`
- `selector-nest-attributes`
- `selector-nest-pseudos`

**Comment by lxsymington** [comment]
@kristerkari would this relate specifically to chaining or is this tied in with combinators as well?

Some of this was implemented as part of #298, perhaps it should not have been? Or was the plan to implement this as exceptions/alternate options?

**Comment by kristerkari** [comment]
> would this relate specifically to chaining or is this tied in with combinators as well?

I was thinking about combinators mainly.

> Some of this was implemented as part of #298, perhaps it should not have been? Or was the plan to implement this as exceptions/alternate options?

My thinking behind the new rules was that you could limit the nesting by the type of the selector. So if you only care that the classes should be nested, then you could enabled that rule.

Now that I'm thinking about this, maybe it would make more sense to introduce these as options to your rule. What would you think about that @lxsymington ?

**Comment by lxsymington** [comment]
@kristerkari, yes I think it would make sense to be part of the same rule. I will try and find some time to have a look at this.

**Comment by mdodge-ecgrow** [comment]
I'm switching over my regular CSS to SCSS.  It's quite a bit of code that needs to switch to nesting.  Is there a rule in this plugin that will show recommend and/or fix all the spots that should be nested?

**Comment by kristerkari** [comment]
@mdodge-ecgrow Unfortunately there is currently no autofix available for that. :(
